### DISCUSSION TITLE
Unable to expand relation named `member`?

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#2678] [Q&A] [answered]
Hi,

Bit of a weird one, I have a `members` auth collections that is the relation target for the `member` field on a `memberships` base collection. I'm unable to expand this relation, field non null and single selection:

```
await pb.collection('memberships').getList(page, perPage, {
    filter: `status = "new" || (status = "active" && updated >= "${yesterday}")`,
    sort: '-updated',
    expand: "member",
    fields: "id,created,updated,expand.member.email"
});
```

Whereas on the same `memberships` collection I have another relation to the `accounts` collection with the field name `account` and this expansion works just fine:

```
await pb.collection('memberships').getList(page, perPage, {
    filter: `status = "new" || (status = "active" && updated >= "${yesterday}")`,
    sort: '-updated',
    expand: "account",
    fields: "id,created,updated,expand.account.email"
});
```

I've checked everything, including `migrate collection` to flatten the migrations. It never expands, full or single fields, could `member` be some sort of reserved keyword?

**Comment by ganigeorgiev** [comment]
Are you sure that your client request satisfies the `member` referenced collection View API rule?

Could you export your collections as JSON (Admin UI > Settings > Export collections)?

**Reply by pnmcosta** [reply]
Ahh that's it! I have `id = @request.auth.id` on both relations, but the request auth is an `accounts` record, hence why it works for accounts and not members, but how could I configure the `members` view rule to also accept the `accounts` auth that have at least one `memberships` for `members`? 💭

**Reply by ganigeorgiev** [reply, marked as answer]
If you don't have direct relation from `members` to the other ones you can use the `@collection.*` rule. There are not enough details from the above example to fully understand your collections structure, but here is a dummy `members` View API rule that checks the `memberships` account:

```js
@request.auth.id != "" &&
@collection.memberships.member ?= id &&
@collection.memberships.account ?= @request.auth.id
```
The above roughly translates to:
- require the request to be authenticated
- require at least one `memberships.member` to match with the current `members.id` and the same matching `memberships` record(s) to have `account` field matching the id of the current authenticated user

**Reply by pnmcosta** [reply]
That's it! I just tweaked it so `members` can view themselves too:

```
id = @request.auth.id ||
(@request.auth.id != "" &&
@collection.memberships.member ?= id &&
@collection.memberships.account ?= @request.auth.id)
```

Many thanks 🙇
